Observer/Config/Save.php

```diff
@@ -98,40 +107,44 @@ public function execute(Observer $observer)
     public function doYotpoApiKeyValidation(Observer $observer)
     {
         $changedPaths = (array)$observer->getEvent()->getChangedPaths();
-        if ($changedPaths && $this->isYotpoKeysChanged($changedPaths)) {
-            $this->cacheTypeList->cleanType(Config::TYPE_IDENTIFIER);
-            $this->appConfig->reinit();
+        if ($changedPaths) {
             $scopeDetails = $this->getScopes($observer);
             $scopeId = (int) $scopeDetails['scope_id'];
-            $scope = $scopeDetails['scope'];
-            $scopes = $scopeDetails['scopes'];
-
-            $appKey = $this->yotpoConfig->getAppKey($scopeId, $scope);
-
-            if ($scope !== ScopeInterface::SCOPE_STORE && !$this->yotpoConfig->isSingleStoreMode()) {
-                $this->resetStoreCredentials($scopeId, $scopes);
-                return true;
-            }
+            if ($this->isYotpoSettingsChanged($changedPaths)) {
+                $this->cacheTypeList->cleanType(Config::TYPE_IDENTIFIER);
+                $this->appConfig->reinit();
+                $scope = $scopeDetails['scope'];
+                $scopes = $scopeDetails['scopes'];
+                $appKey = $this->yotpoConfig->getAppKey($scopeId, $scope);
+
+                if ($scope !== ScopeInterface::SCOPE_STORE && !$this->yotpoConfig->isSingleStoreMode()) {
+                    $this->resetStoreCredentials($scopeId, $scopes);
+                    return true;
+                }
 
-            //Check if appKey is unique:
-            if ($appKey) {
-                foreach ((array) $this->yotpoConfig->getAllStoreIds() as $key => $storeId) {
-                    if (($scopeId && $storeId != $scopeId) && $this->yotpoConfig->getAppKey($storeId) === $appKey) {
-                        $this->resetStoreCredentials($scopeId, $scopes);
-                        throw new AlreadyExistsException(__(
-                            "The APP KEY you've entered is already in use by another store on this system.
-                            Note that Yotpo requires a unique set of APP KEY & SECRET for each store."
-                        ));
+                //Check if appKey is unique:
+                if ($appKey) {
+                    foreach ((array) $this->yotpoConfig->getAllStoreIds() as $key => $storeId) {
+                        if (($scopeId && $storeId != $scopeId) && $this->yotpoConfig->getAppKey($storeId) === $appKey) {
```

Comment:
We should check for `$scopeId` once.
```suggestion
                if ($scopeId && $appKey) {
                    foreach ((array) $this->yotpoConfig->getAllStoreIds() as $key => $storeId) {
                        if ($storeId != $scopeId && $this->yotpoConfig->getAppKey($storeId) === $appKey) {
```

Reply:
This is done